common/scala/src/main/scala/whisk/core/connector/Message.scala

```diff
@@ -53,7 +54,8 @@ case class ActivationMessage(
     activationId: ActivationId,
     activationNamespace: EntityPath,
     content: Option[JsObject],
-    cause: Option[ActivationId] = None)
+    cause: Option[ActivationId] = None,
+    traceMetadata: Option[SpanMetadata] = None)
```

Comment:
Yes streaming payload is a separate concern and issue. The point is to reduce the control plane to just control. Today it does both control and data. There is an issue open to address this. 

On the tracing, again the point is that we will need this information to address the metadata we can relay to clients with activation ids. This means shared state between controller and invoker. The place for such state is redis for example. You don't need to send 1-4KB of data to trace a time delta through a component.

Regardles the cost is both space and time flowing the message over Kafka today. This concerns me, since as I've tried to explain I think there's a more general and better solution that we need to address anyway.